### DISCUSSION TITLE
2.7.0

### DISCUSSION THREAD

**Discussion by arianneorpilla** [arianneorpilla/jidoujisho#217] [Announcements]
This release introduces a **new default card type and standard profile**, with support for **exporting frequencies and proper cloze support**, and miscellaneous improvements related to language and text highlighting.  

**System Requirements**: This application has been tested to work on a variety of devices and emulators running Android 7 to 13. Please note that a significant amount of internal storage space may be taken up by the dictionary database for indexing in order to optimize query speed and performance. Try to give the application around 2-4GB of storage space. Make sure your Android WebView is updated before use. This will resolve common issues that may occur with ッツ Ebook Reader or Mokuro.

# 🔰 New sane defaults
<p align="center" style="margin:0">
<img src="https://i.postimg.cc/HLDrkpzx/Screenshot-20230516-180257.png" width="24%">
<img src="https://i.postimg.cc/CLGgyk5z/Screenshot-20230516-181628-1.png" width="24%">
<img src="https://i.postimg.cc/3xpw6shH/Screenshot-20230516-180323.png" width="24%">
<img src="https://i.postimg.cc/W4v4P5pq/Screenshot-20230516-180445.png" width="24%">
</p>

<p align="center" style="margin:0">
<img src="https://i.postimg.cc/cCHHb6xD/Screenshot-20230516-180358.png" width="24%">
<img src="https://i.postimg.cc/j2tNdyHZ/Screenshot-20230516-181054.png" width="24%">
<img src="https://i.postimg.cc/HxvLy5Xp/Screenshot-20230516-133900.png" width="24%">
<img src="https://i.postimg.cc/P5n5Ydp8/Screenshot-20230516-134038.png" width="24%">
</p>

# 2.7
### 🕳️ Cloze Support
- The new Cloze Before, Inside and After fields allow users to properly export cards with text highlighting.
- When editing Sentence text in the Card Creator, the `Cloze` context menu option will appear when highlighting text in place of `Search`, indicating that making a search will update the cloze fields.
- Users may inspect their cloze fields under `Additional fields` with the new standard profile.
- Please note that manually editing the Sentence field will not affect the cloze fields. Search with the Sentence field, or use the Text Segmentation enhancement to have the Card Creator generate new cloze fields should you wish to update them while editing.
- Selections made in the Text Segmentation enhancement must now be made of consecutively selected items, to enable use for search and selection, enabling the enhancement to update cloze fields appropriately upon use.
- All existing media sources have been updated to provide cloze information for currently highlighted text.
- All existing enhancements used for sentence picking such as ImmersionKit and Massif, have been updated to provide cloze information.

### 🎴 New Card Type and Template
- The `jidoujisho Kinomoto` card type is now the new default card type, with a new accompanying template to easily onboard users with the new fields.
- A new standard profile will be provided upon installation of this version. Existing users that used the `jidoujisho Yuuna` standard profile will be able to keep the legacy standard profile.
- The Context, Cloze Before, Inside and After and Frequency fields are now included and enabled by default in the new standard profile, under `Additional fields`.　
- Users may expand the image in the default card type by tapping on the image.
- The new template now re-orders and organizes together closely related fields at the top, and fields that are unlikely to be manually edited are sorted below.
- Existing users may continue using their legacy profiles, and may simply extend their card types to include more fields.

### 👂 Pitch Accent Diagrams
- The Pitch Accent field now exports with SVG rather than using template-dependent CSS, changes made possible courtesy of **[Immersion Reader](https://github.com/mathewthe2/immersion_reader/blob/main/lib/japanese/draw_pitch.dart)** and **[SVG Pitch](https://github.com/IllDepence/SVG_pitch)**.
- The new pitch accent diagrams will adapt to AnkiDroid's light and dark modes.
- The old CSS-based pitch accent diagrams are now sunset, and existing users will get new pitch accent diagrams with no template changes necessary. 

### ⚙️ Miscellaneous Changes
- Improved text highlighting and search results for English.
- Fixed an issue with sentence picking that caused the app to select the wrong sentence when making text selections on long paragraphs of text.
- Fixed an issue where Instant Export did not appropriately clear media in the Card Creator.
- Removed support for the Text Segmentation enhancement in non-sentence fields.
- Removed the Image Search Term Picker enhancement.
- Duplicate checks now check all card types instead of being limited to the currently selected profile's card type. Note that the first fields of all card types are being used for the check. If you are using a pre-made deck, consider swapping out its card type's first field in order to limit your duplicate checks to your sentence mining card types. 
- The Mokuro media source will now only initially set the `Display two pages` option on the first use of a volume.
- Fixed an issue where changing pages in Mokuro did not respect the `On page turn` option.
- Fixed an issue where the Transcript did not pause when opened with Definition Focus Mode on and Transcript Playback Mode off.
- Fixed an issue where the crop image's grid color is difficult to see in light mode.
- Selecting sentences from Massif and ImmersionKit will now set the source name in the Context field.
- A new option under `Subtitle Timing and Appearance` now allows having the subtitle to always be above the bottom bar area, even if hidden away.

# 📝 The path ahead
> I've rewritten this project a couple of times, with named releases `Chisa` and `Yuuna`, the latter of which is now mature to the extent that I don't think I would ever do such a major rewrite (at least I think so, I'm terrible at taking hiatuses).
>
> Starting with this version, the app will have named releases with major template changes, though it will still internally be named `yuuna`. Cards created even after this version will still include the tag `Yuuna` even if it is from the new card type, and the codebase will remain `yuuna`. 
>
> Card type updates will be named to differentiate new card types from the now deprecated `jidoujisho Yuuna` card type, as I wanted to do better than a simple and mundane numbered naming scheme.
>
>  The `jidoujisho Kinomoto` card type provides what I felt was missing from my app's default card template, and will provide new and existing users with a proper out of the box experience and sane defaults with proper support for exporting with cloze and frequency, and a fresh look for new cards.
>
> I'm still working on the **[wiki](https://github.com/lrorpilla/jidoujisho/wiki)**, and I'll continue working on smaller releases with a focus on bug fixes now that I'm much busier. Anyway, I think that's it for now - I'll save the verbosity for the user and developer documentation that I'm working on!
  
# 📲 Which file should I download?
As a rule of thumb, `arm64` should work for most modern mobile devices. `armeabi` should work on older devices. Generally, desktops and emulators should use `x86_64`.

Depending on which device you are using, download the appropriate `APK` by expanding the `Assets` drop-down below and install on your mobile phone by launching the `APK` with your file explorer.

<hr /><em>This discussion was created from the release <a href='https://github.com/lrorpilla/jidoujisho/releases/tag/2.7.0'>2.7.0</a>.</em>

**Comment by violetu** [comment]


**Reply by arianneorpilla** [reply]
These are very specific non-critical requests, and the app works fine for others for the most part, so I won't action on these over what I believe are critical, but you're always welcome to make a pull request.

I agree that it has multitasking utility, but this suggestion would effectively make part of the screen useless when viewing video, which is the main activity here. 

You don't really need to open the bottom bar even, the problem you're having with the dictionary box clashing with the subtitles doesn't happen if the bottom bar isn't open -- you could always just tap the subtitles directly when the bottom bar is closed, and tap them again to continue playing.

**Reply by arianneorpilla** [reply]
I should clarify my vision with this app is to present media as a native would view -- and replicate that daily use look and not be a very generally customisable video player like VLC.

It's not my intention to make it very customizable like VLC or anything else -- it just needs to present a layout that in which a native or non-learner would normally view video.

I don't intend to offer the user any passive advantages when they don't interact with the screen - when they do not touch anything it should look like a video player normally would -- which it does at present.

The transcript being opened is an explicit gesture that the user needs help, they need to backtrack or get back to a certain line.

I don't provide a side-by-side transcript because I would like users to get comfortable with not having one. I understand that that's a bit contrived on my part, but I am of the belief of making the layout as plain as possible, to keep it simple and keep the focus on watching video like a normal person would, and not be jam-packed of other things.
